server/server.go

```diff
@@ -23,16 +23,25 @@ const (
 	localServerBind = "127.0.0.1:9099"
 )
 
-func StartMetadataServer() error {
+var (
+	metadataRegion string
```

Comment:
Ooof. A global here is a bummer. We might need to rethink this approach.

Reply:
yeah, i wasn't sure. Th server is initialized with this value, and i need somewhere to keep it so it can be used in the http handlers. The handlers are funcs so i can't pass it normally. 

How could we solve this?

Reply:
im being lazy and using this same variable in the "start server" code further down (Line 135). It could be easily passed normally between the functions to avoid using this global

Reply:
I'm going to think on it a bit. We're talking about refactoring this file into some smaller pieces and I want to think about how that might play into things.